src/Pages/Signup/Signup.js

```diff
@@ -0,0 +1,414 @@
+import React, { Component } from 'react';
+import './Signup.scss';
+import Header from '../../Component/HeaderComponent/Header';
+import Footer from '../../Component/FooterComponent/Footer';
+import DaumPostCode from 'react-daum-postcode';
+
+class Signup extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      identity: '',
+      isAvailedId: '',
+      idChecked: false,
+      disabled: true,
+      background: '#d2f7d2',
+      password: '',
+      repassword: '',
+      isAvailedPassword: '',
+      name: '',
+      phone: '',
+      isAvailedPhone: '',
+      email: '',
+      isAvailedEmail: '',
+      // address: '서울시',
+      // zoneCode: '',
+      firstAddress: '',
+      secondAddress: '',
+      isDaumPost: false,
+    };
+  }
+
+  //id 요휴성검사
+  idCheck = e => {
+    const idreg = /^[a-z0-9]{5,15}/g;
+    let userId = e.target.value;
+    this.setState({ identity: e.target.value });
+
+    if (false === idreg.test(userId)) {
+      this.setState(
+        { isAvailedId: '올바른 아이디 형식이 아닙니다.', idChecked: false },
+        () => this.btnChangeColor()
+      );
+    } else {
+      this.setState({ isAvailedId: '', idChecked: true }, () =>
+        this.btnChangeColor()
+      );
+    }
+  };
+  //id 버튼활성화
+  btnChangeColor = () => {
+    this.state.idChecked
+      ? this.setState({ background: '#6ca437', disabled: false })
+      : this.setState({ background: '#d2f7d2', disabled: true });
+  };
+  //id 중복체크
+  rightId = e => {
+    e.preventDefault();
+    fetch('http://10.58.2.234:8000/user/id-check', {
+      method: 'POST',
+      body: JSON.stringify({
+        identity: this.state.identity,
+      }),
+    })
+      .then(response => response.json())
+
+      .then(result => {
+        console.log('결과: ', result);
+        // if (result.message === 'SUCCESS') {
+        //   this.props.history.push('/login');
+        // } else if (result.message === 'IDENTITY_ALREADY_EXISTS') {
+        //   this.setState({ isAvailedId: '이미 존재하는 아이디 입니다.' });
+        // }
+      });
+  };
+
+  //pw 일치여부확인
+  handleConfirmPassword = e => {
+    this.setState({
+      [e.target.name]: e.target.value,
+    });
+    if (e.target.value !== this.state.rePassword) {
+      this.setState({
+        isAvailedPassword: '비밀번호가 일치하지 않습니다.',
+      });
+    } else if (e.target.value === '') {
+      this.setState({
+        isAvailedPassword: '',
+      });
+    } else if (e.target.value === this.state.rePassword) {
+      this.setState({
+        isAvailedPassword: '',
+      });
+    }
```

Comment:
- isAvailedPassword 는 확장성을 고려했을 때 스트링 값 보단, boolean 값으로 관리해주는 것이 더 적합해 보입니다! 
- 확장성 외에도, 좀 더 간결한 코드로 리팩토링 할 수 있게 되기도 합니다!
```suggestion
    this.setState({
      [e.target.name]: e.target.value,
      isAvailedPassword: e.target.value !== this.state.rePassword,
    });
```
- 사실 근데, 이 부분도 결국엔 password, rePassword 스테이트가 업데이트 될 때, 그때그때 계산해주는 것으로 충분히 구현할 수 있기 때문에, 불필요한 메서드 선언인 것 같기도 하네요! 고민해보세요!!

Reply:
비밀번호는 로직이 두개가 겹쳐있어 고민을 좀 해봐야겠네요 ㅠㅠ